### PR TITLE
refactor: split slayer task into monster and kc

### PR DESCRIPTION
**Pull request by iProdigy**
Follow-up from https://github.com/pajlads/DinkPlugin/pull/243#pullrequestreview-1440199599

image

* https://oldschool.runescape.wiki/w/Special:Search?search=Cave%20Kraken

### CODE REVIEW

File: src/main/java/dinkplugin/notifiers/data/SlayerNotificationData.java [hunk 1]
```diff
@@ -2,11 +2,26 @@
...
     String slayerTask;
+
+    @NotNull
     String slayerCompleted;
+
+    @NotNull
     String slayerPoints;
+
+    @Nullable // if jagex changes format of slayerTask
+    Integer incrementalKills;
```
Comment (pajlada) [inline src/main/java/dinkplugin/notifiers/data/SlayerNotificationData.java:22]: double-check if this name is good

**Review by pajlada — APPROVED** [review]
Code looks good, want to double-check naming of the new field but happy to merge

**Review by Felanbird — APPROVED** [review]
image

File: src/main/java/dinkplugin/notifiers/data/SlayerNotificationData.java (hunk 1 above)
Comment (iProdigy) [reply src/main/java/dinkplugin/notifiers/data/SlayerNotificationData.java]: Updated in https://github.com/pajlads/DinkPlugin/pull/248/commits/15afb17ebe3b270e4391236c5ee49f258d416074

**Review by pajlada — APPROVED** [review]
I'm happy with the wording of `killCount` :+1:

**Pull request merged**